index.js

```diff
@@ -10,149 +10,161 @@ function rgbArrToRgbTuples(arr) {
+	palette = quantizer.palette(true, true)
+	// draw it on the canvas
+	canvasCtx.putImageData(new ImageData(new Uint8ClampedArray(quantizedImage), canvas.width, canvas.height), 0, 0);
 
-	function getPixels(px,py) {
-		var pixels = []
-		for (y = 1; y <= PAGE_HEIGHT; y++) {
-			for (x = 1; x <= PAGE_WIDTH; x++) {
-				var onPageIndex = getPageIndex(px,py,x,y);
-				var paletteIndex = paletteImage[onPageIndex] + 1
-				pixels.push(paletteIndex)
-			}
+	processedImageOutput.src = canvas.toDataURL("image/png"); // show the preview
+};
+
+function getPageIndex(px, py, x, y) {
+	return PAGE_WIDTH * pageW * (y + ((py - 1) * PAGE_HEIGHT) - 1) + (px - 1) * PAGE_WIDTH + x - 1
+}
+
+function getPixels(px, py) {
+	var pixels = []
+	for (y = 1; y <= PAGE_HEIGHT; y++) {
+		for (x = 1; x <= PAGE_WIDTH; x++) {
+			var onPageIndex = getPageIndex(px, py, x, y);
+			var paletteIndex = paletteImage[onPageIndex] + 1
+			pixels.push(paletteIndex)
 		}
-		return pixels
 	}
+	return pixels
+}
 
-	function getPalette() {
-		var pal = []
-		for (col of palette) {
-			var color = (col[0]<<16) + (col[1]<<8) + col[2]
-			pal.push(color)
-		}
-		return pal
+function getPalette() {
+	var pal = []
+	for (col of palette) {
+		var color = (col[0] << 16) + (col[1] << 8) + col[2]
+		pal.push(color)
 	}
+	return pal
+}
 
-	const MAX_LABEL_SIZE = 48
+const MAX_LABEL_SIZE = 48
 
-	function lengthLimit(name, info) {
-		if (name.length + info.length > MAX_LABEL_SIZE) {
-			return name.substring(0,MAX_LABEL_SIZE-info.length) + info
-		}
-		return name + info
+function lengthLimit(name, info) {
+	if (name.length + info.length > MAX_LABEL_SIZE) {
+		return name.substring(0, MAX_LABEL_SIZE - info.length) + info
 	}
+	return name + info
+}
 
-	saveButton.onclick = function(event) {
-		var fn = canvas.title.replace(/\.[^/.]+$/, "")
-		var pages = []
-		console.log(pageW,pageH)
-		for (var y = 1; y <= pageH; y++) {
-			for (var x = 1; x <= pageW; x++) {
-				console.log(x,y)
-				pages.push({
-					label: lengthLimit(fn, ' : page ('+ x + ','+ y + ') of (' + pageW + 'x' + pageH + ').'),
-					palette: getPalette(),
-					pixels: getPixels(x,y),
-					width: PAGE_WIDTH,
-					height: PAGE_HEIGHT
-				})
-			}
+saveButton.onclick = function (event) {
+	var fn = canvas.title.replace(/\.[^/.]+$/, "")
+	var pages = []
+	console.log(pageW, pageH)
+	for (var y = 1; y <= pageH; y++) {
+		for (var x = 1; x <= pageW; x++) {
+			console.log(x, y)
+			pages.push({
+				label: lengthLimit(fn, ' : page (' + x + ',' + y + ') of (' + pageW + 'x' + pageH + ')'),
+				palette: getPalette(),
+				pixels: getPixels(x, y),
+				width: PAGE_WIDTH,
+				height: PAGE_HEIGHT
+			})
 		}
-		if (onlyOnePage.checked) {
-			downloadJSON(pages[0], fn+".2dj")
+	}
+	if (onlyOnePage.checked) {
+		downloadJSON(pages[0], fn + ".2dj")
+	} else {
+		if (zippedFile.checked) {
+			var zip = new JSZip();
+			for (var i = 0; i < pages.length; i++) {
+				zip.file(pages[i].label + ".2dj", JSON.stringify(pages[i]));
```

Comment:
`pages[i].label + ".2dj"` results in filenames that are not friendly to type in CC. Maybe something like `fn + "_" + i + ".2dj"` would be more suitable? You loose out on the page information then, but then there aren't spaces in the filename.

Alternatively the labels for the pages could be changed. But spaces don't matter as much in game.

I'm also not sure how much of an impact the spaces would actually have.